src/subsCache.coffee

```diff
@@ -51,22 +86,24 @@ class @SubsCache
       # If we're using fast-render for SSR
       Meteor.subscribe.apply(Meteor.args)
     else
-      hash = EJSON.stringify(args)
+      hash = EJSON.stringify(withoutCallbacks args)
       cache = @cache
       if hash of cache
-        # if we find this subscription in the cache, then restart it
+        # if we find this subscription in the cache, then rescue the callbacks
+        # and restart the cached subscription
+        if hasCallbacks args
+          cache[hash].registerCallbacks callbacksFromArgs args
```

Comment:
I think all we need to do is call onReady here. The onStop callback should still be hooked up from before.